### DISCUSSION TITLE
Convention for the branch naming: maybe needed?

### DISCUSSION THREAD

**Discussion by AndreyDmitriev39r** [jdwilkin4/contrast-ratio-repo#56] [Ideas]
How about we apply some standards for the branch names.
When I have this situation in bash:
```
$ git branch
  53-fix-html-from-validator-report
  create-pr-template
  issue#37-hsl-hsla-contrast-ratio
  issue#40-docs-add-nodejs-prerequisite
  issue#49-typo-readme
* main
  testBranch
```
It's bunch of my branches for my work on issues plus bunch of others branches which I pulled for review purposes
Plus there can be more: my additional branches for working with issues, additional branches for playing around others branches

It can become a bit messy.

Maybe some naming standard for branches can be useful?

**Comment by sparkO7** [comment]
In the freeCodeCamp repo, I can see branches like :
```
chore/japanese-translation-meta-tags

fix/third-party-cypress-action
```

Maybe we can name our branches in the same way.

**Reply by AndreyDmitriev39r** [reply]
This is one way to do it, I like it. But I also like to have in the name number of issue like this:
`53-fix-html-from-validator-report`

**Reply by Kernix13** [reply]
I found a page somewhere that listed various prefixes for branch names based on the purpose of the branch. I know when I was doing typos for FCC RWD Beta course I was using `fix/description_phrase` for my branches. I created a table in the [Pull request title section](https://github.com/Kernix13/beginner-git-commands#pull-request-title) of my Git commands repo for the most common prefixes, but I still am not clear on the best branch name conventions or the best PR titles.

**Reply by AndreyDmitriev39r** [reply]
I am not clear also, but I guess the best one - it's something we should choose(or even invent!) for our team.
From the original post here - what branch name makes most sense? I like this one:
`53-fix-html-from-validator-report`
Tho it can be modified maybe

**Reply by sparkO7** [reply]
I suggest that the label goes first and maybe the issue number at the end?
`fix/html-from-validator-report-53`

**Reply by AndreyDmitriev39r** [reply]
I think that number should go first.
My reasoning:
number >>> it's like id, its unique.
So you can type in bash:
`git checkout 53`
then you hit the TAB
and autocompletion gives you exactly what you need

as for labels - you can have bunch of branches with `fix` at the same time for example

**Reply by sparkO7** [reply]
Oh I see. That is true.

**Comment by jdwilkin4** [comment]
I have read through the discussions and there were some questions around standard conventions for branch names.

Whenever you join a new team, you should first look to the onboarding docs if they exist. 
Some teams will be very particular about what branch names and PR titles they want to use and it will be documented in the onboarding docs for that project.

Other teams are not that picky and will default to some common patterns like these 
```
fix/23,fixing-cypress-tests
chore/setting-up-prettier-config
updating-readme
```
It is fine to come up with your own pattern as long as the whole team agrees.

**Comment by jdwilkin4** [comment]
> It can become a bit messy.

You can delete branches locally once they have been merged into `main`.
That will clean up your branch list.

```
git branch -d localBranchName
```

**Reply by AndreyDmitriev39r** [reply]
Oh yeah I am deleting branches regularly. But still, sometimes I have like a several branches which I don't want to delete at the moment
